### ISSUE TITLE
Daemon terminates immediately after launching

### ISSUE THREAD

**Issue by DanTup** [dart-lang/webdev#215]
When running the daemon (using code from https://github.com/dart-lang/webdev/pull/212), it builds and launches Chrome, but then immediately terminates.

```
[{"event":"app.start","params":{"appId":"Pg0KOLVX/N8ilJZYuu+HqA==","directory":"/Users/dantup/Dev/foo/examples/hello_world","deviceId":"chrome","launchMode":"run"}}]
Unhandled exception:
Uncaught, null, 85, 0, object null
#0      _handleErrorIfPresent (package:dwds/src/chrome_proxy_service.dart:752:5)
#1      ChromeProxyService.createIsolate (package:dwds/src/chrome_proxy_service.dart:114:5)
<asynchronous suspension>
#2      ChromeProxyService.create (package:dwds/src/chrome_proxy_service.dart:79:19)
<asynchronous suspension>
#3      DebugService.start (package:dwds/service.dart:68:34)
<asynchronous suspension>
#4      DevHandler.startDebugService (package:webdev/src/serve/handlers/dev_handler.dart:65:25)
<asynchronous suspension>
#5      AppDomain._initialize (package:webdev/src/daemon/app_domain.dart:39:26)
<asynchronous suspension>
#6      new AppDomain (package:webdev/src/daemon/app_domain.dart:68:5)
#7      DaemonCommand.run (package:webdev/src/command/daemon_command.dart:63:27)
<asynchronous suspension>
#8      CommandRunner.runCommand (package:args/command_runner.dart:196:27)
<asynchronous suspension>
#9      _CommandRunner.runCommand (package:webdev/src/webdev_command_runner.dart:38:24)
<asynchronous suspension>
#10     CommandRunner.run.<anonymous closure> (package:args/command_runner.dart:111:29)
#11     new Future.sync (dart:async/future.dart:224:31)
#12     CommandRunner.run (package:args/command_runner.dart:111:11)
#13     run (package:webdev/src/webdev_command_runner.dart:19:56)
#14     main (file:///Users/dantup/.pub-cache/git/webdev-564594a0001da49f8a39986885113ee25776746a/webdev/bin/webdev.dart:17:22)
<asynchronous suspension>
#15     _startIsolate.<anonymous closure> (dart:isolate/runtime/lib/isolate_patch.dart:298:32)
#16     _RawReceivePortImpl._handleMessage (dart:isolate/runtime/lib/isolate_patch.dart:171:12)
```

I don't know if these errors in the console are the cause, or just a symptom of the server having gone away.

<img width="1201" alt="Screenshot 2019-03-18 at 5 42 52 pm" src="https://user-images.githubusercontent.com/1078012/54551677-70f4c900-49a6-11e9-9c4a-b23802793f2e.png">

**Comment by grouma** [comment]
The errors in the console look like a result of the server dying.

What is your pubspec? Also what version of the SDK are you running? I'd like to try and reproduce this locally.

**Comment by DanTup** [comment]
@grouma My pubspec is the hello_world example project, but I added `webdev` as a `dev_dependency` and added this to `dependency_overrides`:

``` yaml
  dwds:
    git:
      url: https://github.com/dart-lang/webdev.git
      path: dwds
      ref: log-message
  webdev:
    git:
      url: https://github.com/dart-lang/webdev.git
      path: webdev
      ref: log-message
```

In VS Code, I'm trying to avoid relying on globally activated packages as much as possible and use `pub global run`, because it's more flexible (allowing different versions). I don't know if it's going to work out here, but I figured I'd start that way anyway.

I'm then running it like this from the `hello_world` folder:

```
~/Dev/dart-sdk/nightly-2019-03-14/bin/pub run webdev daemon
```

(running from the terminal has the same behaviour).

The SDK I'm using is:

Dart VM version: 2.2.1-edge.1af9d3e14b0f643b22151a006a12ddbd6a33ca1b (Thu Mar 14 12:35:56 2019 +0000) on "macos_x64"

Ping me if you can't repro!

**Comment by DanTup** [comment]
Doesn't happen if I put this version of the SDK on `PATH`, so I think this will be fixed by #216.

**Comment by DanTup** [comment]
(Closing as #216 has landed)
